hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -350,13 +350,24 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		if hcluster.Spec.Platform.AWS == nil {
 			return ctrl.Result{}, fmt.Errorf("the HostedCluster for this NodePool has no .Spec.Platform.AWS, this is unsupported")
 		}
+		if nodePool.Spec.Arch == "" {
+			nodePool.Spec.Arch = "x86_64"
+		}
+		if nodePool.Spec.Platform.AWS.InstanceType == "" {
+			switch nodePool.Spec.Arch {
+			case "x86_64":
+				nodePool.Spec.Platform.AWS.InstanceType = "m5.large"
```

Comment:
Please let's drop this, we don't want to default self updating the spec in the backend. This is API required input, the cli can automate convenient input.

Reply:
@enxebre do you want lines 353 to 355 removed also?